ui/lib/config-ui/addon/components/messages/message-expiration-date-form.js

```diff
@@ -20,14 +21,33 @@ import { datetimeLocalStringFormat } from 'core/utils/date-formatters';
 export default class MessageExpirationDateForm extends Component {
   datetimeLocalStringFormat = datetimeLocalStringFormat;
   @tracked groupValue = 'never';
-  @tracked formDateTime = '';
+  @tracked messageEndTime = '';
 
   constructor() {
     super(...arguments);
 
     if (this.args.message.endTime) {
       this.groupValue = 'specificDate';
-      this.formDateTime = this.args.message.endTime;
+      this.messageEndTime = this.args.message.endTime;
     }
   }
+
+  get validationError() {
+    const validations = this.args.modelValidations || {};
+    const state = validations[this.args.attr.name];
+    return state && !state.isValid ? state.errors.join(' ') : null;
+  }
+
+  @action
+  specificDateChange() {
+    this.groupValue = 'specificDate';
+    this.args.message.endTime = this.messageEndTime;
+  }
+
+  @action
+  onFocusOut(e) {
+    this.messageEndTime = e.target.value;
+    this.args.message.endTime = this.messageEndTime;
```

Comment:
Curious why you are tracking the `endTime` with a local variable rather than directly mutating the property on `args.message`?

Reply:
We'll need some way to remember the old `endTime` that was selected. If a user selects a specific date, and decides to click 'never', but then goes back to select specific date they'll have their previous state.